compiler/src/dotty/tools/dotc/transform/Constructors.scala

```diff
@@ -79,8 +93,7 @@ class Constructors extends MiniPhaseTransform with IdentityDenotTransformer { th
   }
 
   override def transformValDef(tree: tpd.ValDef)(implicit ctx: Context, info: TransformerInfo): tpd.Tree = {
-    if (mightBeDropped(tree.symbol))
-      (if (isWildcardStarArg(tree.rhs)) retainedPrivateVals else seenPrivateVals) += tree.symbol
```

Comment:
It was dead code. The test looks non-sensical to me. I fail to see how a definition of a class member can end up as something like this:

     val xs = y: _*

Even if it did, why would it make a difference? I put in an assert to flag the consition, but never triggered in all our tests.